scn/post_upload_s3.bash

```diff
@@ -0,0 +1,123 @@
+		echo "checksum failure"
+		tmpfile=/tmp/dcm_fail-$$.json
+		echo "{\"status\":\"validation failed\",\"uploadFolder\":\"${ulidFromJson}\",\"totalSize\":0}" > $tmpfile 
+		curl --insecure -H "X-Dataverse-key: ${DVAPIKEY}" -H "Content-type: application/json" -X POST --upload-file $tmpfile "${DVHOST}/api/datasets/:persistentId/dataCaptureModule/checksumValidation?persistentId=doi:${DOI_SHOULDER}/${ulidFromJson}"
+	else
+		# handle checksum success
+		echo "checksums verified"
+
+		#move to HOLD location
+		
+		if [ ! `aws s3 ls s3://${S3HOLD}/${ulidFromJson}/`]; then    #this check is different than normal post_upload, we don't use the extra folder level
+			aws s3 cp --recursive ${DEPOSIT}/${ulidFolder}/${ulidFolder}/ s3://${S3HOLD}/${ulidFromJson}/ #this does not copy empty folders from DEPOSIT as folders do not actually exist in s3
+
+			err=$?
+			if (( $err != 0 )) ; then
+				echo "dcm: file move $ulid" 
+				break
+			fi
+			rm -rf ${DEPOSIT}/${ulidFolder}/${ulidFolder}
+			echo "data moved"
+			tmpfile=/tmp/dcm_fail-$$.json # not caring that the success tmp file has "fail" in the name
+
+			#This may prove to be slow with large datasets
+			sz=`aws s3 ls --summarize --human-readable --recursive s3://${S3HOLD}/${ulidFromJson}/ | grep "Total Size: " | cut -d' ' -f 6`
+
+			echo "{\"status\":\"validation passed\",\"uploadFolder\":\"${ulidFromJson}\",\"totalSize\":$sz}" > $tmpfile 
+
+			dvr=`curl -s --insecure -H "X-Dataverse-key: ${DVAPIKEY}" -H "Content-type: application/json" -X POST --upload-file $tmpfile "${DVHOST}/api/datasets/:persistentId/dataCaptureModule/checksumValidation?persistentId=doi:${DOI_SHOULDER}/${ulidFromJson}"`
+			dvst=`echo $dvr | jq -r .status`
+
+			if [ "OK" != "$dvst" ]; then
+				#TODO - this block should email alerts queue
+				echo "ERROR: dataverse at ${DVHOST} had a problem handling the DCM success API call"
+				if [ "DVHOST" == "${DVHOST}" ]; then
+					echo "DVHOST == DVHOST; assuming internal test without Dataverse and not retrying"
+				else
+					echo "$dvr"
+					echo "will retry in $retry_delay seconds"
+					sleep $retry_delay
+					dvr_rt=`curl -s --insecure -H "X-Dataverse-key: ${DVAPIKEY}" -H "Content-type: application/json" -X POST --upload-file $tmpfile "${DVHOST}/api/datasets/:persistentId/dataCaptureModule/checksumValidation?persistentId=doi:${DOI_SHOULDER}/${ulidFromJson}"`
+					dvst_rt=`echo $dvr | jq -r .status`
+					if [ "OK" != "$dvst_rt" ]; then
+						echo "ERROR: retry failed, will need to handle manually"
+					else
+						echo "ERROR: retry succeeded"
+					fi
+				fi
+			fi
+		else
+			echo "handle error - duplicate upload id $ulidFolder"
+			echo "problem moving data; bailing out"
+			#TODO - dv isn't listening for this error condition
+			break #FIXME - this breaks out of the loop; aborting the scan (instead of skipping this dataset)
+		fi
+
+		aws s3 cp $DEPOSIT/processed/${ulidFolder}.json s3://${S3HOLD}/stage/
```

Comment:
Having these files on S3 is likely to be inefficient (potentially large number of small files); it may make more sense to store them somewhere else on the POSIX filesystem, or to eliminate them once a transfer has been verified.